apps/meteor/client/views/teams/contextualBar/channels/hooks/useToggleAutoJoin.tsx

```diff
@@ -1,18 +1,41 @@
 import type { IRoom } from '@rocket.chat/core-typings';
-import { useEndpoint, usePermission, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useEndpoint, usePermission, useSetting, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { t } from 'i18next';
 
-export const useToggleAutoJoin = (room: IRoom, { reload }: { reload?: () => void }) => {
+import { roomCoordinator } from '../../../../../lib/rooms/roomCoordinator';
+
+export const useToggleAutoJoin = (room: IRoom, { reload, mainRoom }: { reload?: () => void; mainRoom: IRoom }) => {
 	const dispatchToastMessage = useToastMessageDispatch();
 	const updateRoomEndpoint = useEndpoint('POST', '/v1/teams.updateRoom');
 	const canEditTeamChannel = usePermission('edit-team-channel', room._id);
+	const maxNumberOfAutoJoinMembers = useSetting<number>('API_User_Limit');
 
 	const handleToggleAutoJoin = async () => {
+		// Sanity check, the setting has a default value, therefore it should always be defined
+		if (!maxNumberOfAutoJoinMembers) {
+			return;
+		}
+
 		try {
-			await updateRoomEndpoint({
+			const { room: updatedRoom } = await updateRoomEndpoint({
 				roomId: room._id,
 				isDefault: !room.teamDefault,
 			});
-			dispatchToastMessage({ type: 'success', message: room.teamDefault ? 'channel set as non autojoin' : 'channel set as autojoin' });
+			if (updatedRoom.teamDefault) {
+				// If the number of members in the mainRoom (the team) is greater than the limit, show an info message
+				// informing that not all members will be auto-joined to the channel
+				const messageType = mainRoom.usersCount > maxNumberOfAutoJoinMembers ? 'info' : 'success';
+				const message = mainRoom.usersCount > maxNumberOfAutoJoinMembers ? 'Team_Auto-join_exceeded_user_limit' : 'Team_Auto-join_updated';
```

Comment:
Passing `mainRoom` through the tree just to get `usersCount` seems a bit wasteful to me. It almost feels like that information should be returned by the endpoint performing the action (`teams.updateRoom`) 🤔

Reply:
This value is the number of users inside the team, we can use it to tell whether we will be able to `Auto-Join` them all if the `mainRoom.usersCount` is greater than the limit.
There is no way to that in the teams.updateRoom endpoint without having to do additional queries just to get the value of the usersCount of the team.
Me and @matheusbsilva137 thought some prop drilling was worth it to not have to query the database for that data.